superset-frontend/src/components/Chart/Chart.jsx

```diff
@@ -232,16 +246,50 @@ class Chart extends React.PureComponent {
     );
   }
 
+  renderSpinner(databaseName) {
+    const message =
+      databaseName === undefined
+        ? t('Waiting on database...')
+        : t('Waiting on %s', databaseName);
```

Comment:
Let's do it like this since it gives more context for the translation:

```suggestion
    const message =
      databaseName
        ? t('Waiting on %s', databaseName)
        : t('Waiting on database...');
```